src/broker/saslAuthenticator/awsIam.js

```diff
@@ -10,18 +10,16 @@ module.exports = class AWSIAMAuthenticator {
 
   async authenticate() {
     const { sasl } = this.connection
-    if (!sasl.authorizationIdentity) {
+    const props = sasl
```

Comment:
Is there a reason that you are re-assigning `sasl` to this `props` variable? I can't see any reason to do it, so please remove it and just use `sasl` directly.